vyper/semantics/analysis/module.py

```diff
@@ -789,7 +790,13 @@ def _add_import(
     # load an InterfaceT or ModuleInfo from an import.
     # raises FileNotFoundError
     def _load_import(self, node: vy_ast.VyperNode, level: int, module_str: str, alias: str) -> Any:
-        return self._load_import_helper(node, level, module_str, alias)
+        # the directory this (currently being analyzed) module is in
+        self_search_path = Path(self.ast.resolved_path).parent
+
+        with self.input_bundle.search_path(
```

Comment:
i think for a relative import, we actually need to override the entire search path. can be a context manager, but something along the lines of
```
try:
    self.input_bundle.search_paths = [self_search_path]
    self._load_import_helper(...)
finally:
    self.input_bundle.search_paths = original_search_paths
```

otherwise you can break out of the current directory if the search path falls back to some other path (e.g. like /usr/lib/site-packages/foo/bar)

Reply:
The proposed solution is similar - it's just tricky when we have a file with a relative path import that imports something with an absolute path. Inside that file, we need all the search paths (which would be deleted by the previous relative import). Therefore, the new solution keeps all the search paths and switches between searching only the topmost path and all the paths.

Reply:
Tricky indeed! I'm not sure I like changing the InputBundle api just for this, it seems like it should rather be part of the recursion state.

Reply:
True true!  and there is another problem with the solution now  - the "current search paths" are appended to the list, so an absolute import could look into a "current search path" of a previous relative import.